src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -225,8 +227,41 @@ public CodeGenerator getCodeGenerator(Format outputFormat, CodeConsumer cc) {
       }
     }
 
+    result.moduleRewriteLog = generateModuleRewriteLog(filesToConvert, modulePrePass);
     ((GentsErrorManager) compiler.getErrorManager()).doGenerateReport();
-    return sourceFileMap;
+    return result;
+  }
+
+  /**
+   * Generates a log file that maps how goog.modules/goog.provides where mapped to TS modules.
```

Comment:
Have you considered making the log structured data, e.g. a JSON structure? That'd make it easier to parse for many downstream tools, easier to understand (self describing), and easier to detect errors. Plus you save a on a bunch of escaping nightmares (e.g. files containing commas). Also makes for easier extensibility if we need more metadata down the line.

```
[
  {originalNamespace: "...", tsPath: "...", defaultExport: "..."}.
  ...
]
```